src/main/java/com/programmers/repository/FileBookRepository.java

```diff
@@ -0,0 +1,101 @@
+package com.programmers.repository;
+
+import com.programmers.domain.Book;
+import com.programmers.domain.BookState;
+import com.programmers.provider.BookIdProvider;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class FileBookRepository implements BookRepository {
+    private final List<Book> books = new ArrayList<>();
+    private final String csvFile = "./app/src/main/resources/data.csv";
+
+    private FileBookRepository() {
+    }
+
+    private static class Holder {
+        private static final FileBookRepository INSTANCE = new FileBookRepository();
+    }
+
+    public static FileBookRepository getInstance() {
+        FileBookRepository fileBookRepository = FileBookRepository.Holder.INSTANCE;
+        fileBookRepository.loadDataFromFile();
+        return fileBookRepository;
+    }
+
+    @Override
+    public void addBook(Book book) {
+        books.add(book);
+        updateFile();
+    }
+
+    @Override
+    public List<Book> getAllBooks() {
+        return books;
+    }
+
+    @Override
+    public Optional<Book> findBookById(int id) {
+        return books.stream().filter(book -> book.getId() == id).findAny();
```

Comment:
```suggestion
        return books.stream()
        .filter(book -> book.getId() == id)
        .findAny();
```
1. 한줄의 점 하나씩으로 좀더 위에서 아래로 읽히는 방식으로 하면 어떨까요?
2. book에게 이 id와 맞아? 즉, isSameId와 같이 메시지를 보내서 물어보면 어떨까요?
3. findAny vs findFirst에 대해서 한번 알아보시겠어요?

Reply:
감사합니다! 말씀해주신 방법을 적용하여 가독성을 개선하였고, findAny와 findFirst의 차이에 대해 알아보았습니다.
findAny의 경우 순서가 보장되지 않고 병렬 스트림에서 주로 사용된다는 것에 대해서도 알게 되었습니다.

id가 unique한 값이기 때문에 어떤 메소드를 사용해도 결과값은 같겠지만, 이 프로젝트에서는 병렬스트림까지 이용할 필요가 없다고 생각되어 사용 방법만 익힌 뒤 findFirst를 적용하였습니다.